numpy/core/src/private/ufunc_override.h

```diff
@@ -153,6 +154,64 @@ normalize_at_args(PyUFuncObject *ufunc, PyObject *args,
 }
 
 /*
+ * Check args for a object with a `__numpy_ufunc__` attribute.
+ */
+static int
+check_tuple_for_numpy_ufunc(PyObject *args, int *noa,
+                            int *i, PyObject *with_override[NPY_MAXARGS],
+                            int with_override_pos[NPY_MAXARGS]) {
+    int og_i = *i; /* original i */
+    int og_noa = *noa; /* original number of overriding args */
+    int nargs = PyTuple_GET_SIZE(args);
+    PyObject *obj;
+
+    for (; *i < (og_i + nargs); ++*i) {
```

Comment:
Ewww. Instead of `*i`, use a local variable and assign its value on exit. Best rename the argument to something other than `i` also.